camel/agents/chat_agent.py

```diff
@@ -359,20 +398,33 @@ def step(
                 )
                 break
 
+        # if use structure response, set structure result as content of
+        # BaseMessage
+        if output_schema:
+            for base_message_item in output_messages:
+                base_message_item.content = str(info['tool_calls'][-1].result)
```

Comment:
will open-source models also go through this? I think open-source models don't support tool callings right